### PR TITLE
fix IAM policy set

### PR DESCRIPTION
**Pull request by kunmingg**
fix https://github.com/kubeflow/kubeflow/issues/2659

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2681)

### CODE REVIEW

File: bootstrap/pkg/utils/iamutils.go [hunk 1]
```diff
@@ -82,6 +69,31 @@ func GetIamPolicy(project string) (*cloudresourcemanager.Policy, error) {
 	return service.Projects.GetIamPolicy(project, req).Context(ctx).Do()
 }
 
+// Gets IAM plicy from GCP for the whole project.
```
Comment (lluunn) [inline bootstrap/pkg/utils/iamutils.go]: what does cleared mean? Could you explain in comment?

File: bootstrap/pkg/utils/iamutils.go [hunk 2]
```diff
@@ -82,6 +69,31 @@ func GetIamPolicy(project string) (*cloudresourcemanager.Policy, error) {
...
+func GetClearIamPolicy(currentPolicy *cloudresourcemanager.Policy, pendingPolicy *cloudresourcemanager.Policy) *cloudresourcemanager.Policy {
```
Comment (lluunn) [inline bootstrap/pkg/utils/iamutils.go]: GetClearedIamPolicy?

File: bootstrap/pkg/utils/iamutils.go (hunk 1 above)
Comment (kunmingg) [reply bootstrap/pkg/utils/iamutils.go]: updated

**Comment by jlewi** [comment]
Would you mind adding a unittest to verify that ClearedIamPolicy returns the expected policy? We'd like to verify we don't delete accounts that shouldn't be deleted.

**Comment by kunmingg** [comment]
@jlewi 
updated

File: bootstrap/pkg/utils/iamutils_test.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+func Test(t *testing.T) {
+	tests := []struct {
```
Comment (jlewi) [inline bootstrap/pkg/utils/iamutils_test.go]: Thanks. So the typical style in go is to define a list of testCases where each testCase would specify the input and the expected output.

Would it be possible to write the test that way?

e.g. add a struct variable that should equal the expectedPolicy returned by GetClearediamPolicy. Then do a deepEqual to compare the expected and actual. 

Something like:
https://github.com/kubeflow/tf-operator/blob/cf01b5125e25239042e1caef1ef44f7aaf2337e3/pkg/apis/tensorflow/v1beta1/defaults_test.go#L117

Comment (kunmingg) [reply bootstrap/pkg/utils/iamutils_test.go]: updated

**Comment by jlewi** [comment]
/lgtm
/approve
Thanks @kunmingg

**Pull request merged**